Prox/Prox/PlaceCarousel/PlaceCarouselViewController.swift

```diff
@@ -16,6 +16,16 @@ protocol PlaceDataSource: class {
     func index(forPlace: Place) -> Int?
     func fetchPlace(placeKey: String, withEvent eventKey: String, callback: @escaping (Place?) -> ())
     func sortPlaces(byLocation location: CLLocation)
+
+    /// Returns all nearby places filtered by the given set of filters.
+    /// This function does not modify the data source.
+    func filterPlaces(filters: [PlaceFilter]) -> [Place]
+
+    /// Refreshes the data source by filtering with the enabled filters.
+    func refresh()
+
+    /// You must call refresh() after changing the enabled state of any filter!
```

Comment:
Not quite sure what you're suggesting here. `PlacesProvider` has multiple filters, so what would your proposed `enabled` field look like? Would it take an array of the same cardinality as the filter array, with correlated indices?